File: app/src/main/java/me/ykrank/s1next/view/adapter/PostListRecyclerViewAdapter.java
package me.ykrank.s1next.view.adapter;

import android.content.Context;
import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.recyclerview.widget.RecyclerView;

import com.github.ykrank.androidtools.ui.adapter.delegate.FooterProgressAdapterDelegate;

import me.ykrank.s1next.data.api.model.Thread;
import me.ykrank.s1next.data.api.model.Vote;
import me.ykrank.s1next.view.adapter.delegate.PostAdapterDelegate;

/**
 * This {@link RecyclerView.Adapter}
 * has another item type {@link FooterProgressAdapterDelegate}
 * in order to implement pull up to refresh.
 */
public final class PostListRecyclerViewAdapter extends BaseRecyclerViewAdapter {
    @NonNull
    private PostAdapterDelegate postAdapterDelegate;

    public PostListRecyclerViewAdapter(Fragment fragment, @NonNull Context context) {
        super(context, true);

        postAdapterDelegate = new PostAdapterDelegate(fragment, context);
        addAdapterDelegate(postAdapterDelegate);
    }

    public void setThreadInfo(@NonNull Thread threadInfo, int pageNum) {
        postAdapterDelegate.setThreadInfo(threadInfo, pageNum);
    }

    public void setVoteInfo(@Nullable Vote voteInfo) {
        postAdapterDelegate.setVoteInfo(voteInfo);
    }
}

File: app/src/main/java/me/ykrank/s1next/view/page/post/adapter/PostListRecyclerViewAdapter.kt
package me.ykrank.s1next.view.page.post.adapter

import android.content.Context
import androidx.fragment.app.Fragment
import me.ykrank.s1next.data.api.model.Thread
import me.ykrank.s1next.data.api.model.Vote
import me.ykrank.s1next.view.adapter.BaseRecyclerViewAdapter

/**
 * This [RecyclerView.Adapter]
 * has another item type [FooterProgressAdapterDelegate]
 * in order to implement pull up to refresh.
 */
class PostListRecyclerViewAdapter(fragment: Fragment, context: Context) :
    BaseRecyclerViewAdapter(context, true) {
    private val postAdapterDelegate = PostAdapterDelegate(fragment, context)
    private val postBlackAdapterDelegate = PostBlackAdapterDelegate(fragment, context)

    init {
        addAdapterDelegate(postAdapterDelegate)
        addAdapterDelegate(postBlackAdapterDelegate)
    }

    fun setThreadInfo(threadInfo: Thread, pageNum: Int) {
        postAdapterDelegate.setThreadInfo(threadInfo, pageNum)
    }

    fun setVoteInfo(voteInfo: Vote?) {
        postAdapterDelegate.setVoteInfo(voteInfo)
    }
}
